rtlib/src/ImageStitcher.cpp

```diff
@@ -0,0 +1,732 @@
+        std::pair<float, float> point2;
+        // Read in the landmarks and store them
+        landmarkFile >> point1.first;
+        while (!landmarkFile.eof()) {
+            landmarkFile >> point1.second;
+            ldm.src_ldms.push_back(point1);
+            landmarkFile >> point2.first;
+            landmarkFile >> point2.second;
+            ldm.dst_ldms.push_back(point2);
+            landmarkFile >> point1.first;
+        }
+        // Close the landmarks file
+        landmarkFile.close();
+        landmarks_.emplace_back(ldm);
+    }
+}
+
+std::vector<cv::detail::ImageFeatures> ImageStitcher::find_features_(double& work_scale_) {
+    cv::Ptr<cv::detail::FeaturesFinder> features_finder_ = new cv::detail::OrbFeaturesFinder();
+    std::vector<cv::UMat> feature_find_imgs(imgs_.size());
+    std::vector<cv::UMat> feature_find_masks(masks_.size());
+
+    std::vector<cv::detail::ImageFeatures> features_(imgs_.size());
+
+    for (size_t i = 0; i < imgs_.size(); ++i) {
+            cv::resize(
+                    imgs_[i], feature_find_imgs[i], cv::Size(), work_scale_,
+                    work_scale_, cv::INTER_LINEAR);
+
+
+        if (!masks_.empty()) {
+            cv::resize(
+                    masks_[i], feature_find_masks[i], cv::Size(), work_scale_,
+                    work_scale_, cv::INTER_NEAREST);
+        }
+        features_[i].img_idx = (int)i;
+        // LOGLN("Features in image #" << i+1 << ": " <<
+        // features_[i].keypoints.size());
+    }
+
+    // find features possibly in parallel
+    // if (rois_.empty())
+
+    (*features_finder_)(feature_find_imgs, features_);
+
+    // else
+    //(*features_finder_)(feature_find_imgs, features_, feature_find_rois);
+
+    // Do it to save memory
+    feature_find_imgs.clear();
+    feature_find_masks.clear();
+
+    return features_;
+}
+
+void ImageStitcher::find_matches_(double conf_thresh_, std::vector<cv::UMat>& seam_est_imgs_, std::vector<cv::Size>& full_img_sizes_) {
+    std::vector<cv::detail::MatchesInfo> pairwise_matches_;
+    cv::Ptr<cv::detail::FeaturesMatcher> features_matcher_ = cv::makePtr<cv::detail::AffineBestOf2NearestMatcher>(false, false);
+    cv::UMat matching_mask_;
+    // LOGLN("Finding features, time: " << ((getTickCount() - t) /
+    // getTickFrequency()) << " sec");
+
+    (*features_matcher_)(all_features_, all_pairwise_matches_, matching_mask_);
+    features_matcher_->collectGarbage();
+    // LOGLN("Pairwise matching, time: " << ((getTickCount() - t) /
+    // getTickFrequency()) << " sec");
+
+    // This is for option 1 with placing the matches before filtering the images
+    if (option_ == 1 && !landmarks_.empty()){
+        for(int i = 0; i < landmarks_.size(); i++){
+            insert_user_matches_(landmarks_[i]);
+        }
+    }
+    // Leave only images we are sure are from the same panorama
+    auto indices_ = cv::detail::leaveBiggestComponent(
+            all_features_, all_pairwise_matches_, (float)conf_thresh_);
```

Comment:
Why is the confidence threshold a double if it's used as a float?

Reply:
In the bundle adjuster, it uses `conf_thresh_` as a double. Should I change it to a float?